lib/plugin-api/hydrogen-kernel.js

```diff
@@ -27,6 +78,70 @@ export default class HydrogenKernel {
     }
   }
 
+  /*
+   * The language of the kernel, as specified in its kernelspec
+   */
+  get language(): string {
+    this._assertNotDestroyed();
+    return this._kernel.language;
+  }
+
+  /*
+   * The display name of the kernel, as specified in its kernelspec
+   */
+  get displayName(): string {
+    this._assertNotDestroyed();
+    return this._kernel.displayName;
+  }
+
+  /*
+   * Add a kernel middleware, which allows intercepting and issuing commands to
+   * the kernel.
+   */
+  addMiddleware(middleware: HydrogenKernelMiddleware) {
+    this._assertNotDestroyed();
+
+    const prevMiddlewareStack = this._kernel.middlewareStack;
+    const middlewareStack = Object.assign({}, prevMiddlewareStack);
+
+    // The middleware object we get as an argument may later be mutated to
+    // remove/replace some of its methods. The current plugin API is designed to
+    // ignore any such changes, which requires making a copy here. We use local
+    // variables instead of copying the entire object to make flow happy.
+    if (!!middleware.interrupt) {
```

Comment:
Out of curiosity, is there an advantage to the double negative here?

Reply:
Actually, I have no idea about this. I'm not very familiar with the edge cases of javascript typecasting.